contrib/registry/eureka/client.go

```diff
@@ -0,0 +1,363 @@
+	heartbeatRetry      = 3
+)
+
+const tpl = `{
+  "instance": {
+	"instanceId": "{{ .InstanceID }}",
+    "hostName": "{{ .IP }}",
+    "app": "{{ .AppID }}",
+    "ipAddr": "{{ .IP }}",
+    "vipAddress": "{{ .AppID }}",
+	"overriddenstatus": "UNKNOWN",
+    "status":"UP",
+    "port": {
+      "$":{{ .Port }},
+      "@enabled": true
+    },
+    "securePort": {
+      "$":{{ .SecurePort }},
+      "@enabled": false
+    },
+    "homePageUrl" : "{{ .HomePageURL }}",
+    "statusPageUrl": "{{ .StatusPageURL }}",
+    "healthCheckUrl": "{{ .HealthCheckURL }}",
+    "dataCenterInfo" : {
+      "@class":"com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo",
+      "name": "MyOwn"
+    },
+    "metadata": {
+			{{- range $key, $value := .MetaData }}
+			"{{ $key }}": "{{ $value }}",
+	 		{{- end }}
+			"agent" : "go-eureka-client"
+    }
+  }
+}`
+
+type Endpoint struct {
+	InstanceID     string
+	IP             string
+	AppID          string
+	Port           string
+	SecurePort     string
+	HomePageURL    string
+	StatusPageURL  string
+	HealthCheckURL string
+	MetaData       map[string]string
+}
+
+// Response for /eureka/apps
+type ApplicationsRootResponse struct {
+	ApplicationsResponse `json:"applications"`
+}
+
+type ApplicationsResponse struct {
+	Version      string        `json:"versions__delta"`
+	AppsHashcode string        `json:"apps__hashcode"`
+	Applications []Application `json:"application"`
+}
+
+type Application struct {
+	Name     string     `json:"name"`
+	Instance []Instance `json:"instance"`
+}
+
+type Instance struct {
+	InstanceID         string            `json:"instanceId"`
+	HostName           string            `json:"hostName"`
+	Port               Port              `json:"port"`
+	App                string            `json:"app"`
+	IPAddr             string            `json:"ipAddr"`
+	Status             string            `json:"status"`
+	SecurePort         SecurePort        `json:"securePort"`
+	LastDirtyTimestamp string            `json:"lastDirtyTimestamp"`
+	Metadata           map[string]string `json:"metadata"`
+}
+
+type Port struct {
+	Port int `json:"$"`
+}
+
+type SecurePort struct {
+	SecurePort int `json:"$"`
+}
+
+var _ APIInterface = new(Client)
+
+type APIInterface interface {
+	Register(ctx context.Context, ep Endpoint) error
+	Deregister(ctx context.Context, appID, instanceID string) error
+	Heartbeat(ep Endpoint)
+	FetchApps(ctx context.Context) []Application
+	FetchAllUpInstances(ctx context.Context) []Instance
+	FetchAppInstances(ctx context.Context, appID string) (m Application, err error)
+	FetchAppUpInstances(ctx context.Context, appID string) []Instance
+	FetchAppInstance(ctx context.Context, appID string, instanceID string) (m Instance, err error)
+	FetchInstance(ctx context.Context, instanceID string) (m Instance, err error)
+	Out(ctx context.Context, appID, instanceID string) error
+	Down(ctx context.Context, appID, instanceID string) error
+}
+
+type Opt func(e *Client)
```

Comment:
It is better to unify the code style，https://github.com/go-kratos/kratos/blob/9662ef3c213d3f7802c27d6b85386bd649a76de7/contrib/registry/etcd/registry.go#L19 for etcd, https://github.com/go-kratos/kratos/blob/9662ef3c213d3f7802c27d6b85386bd649a76de7/contrib/registry/nacos/registry.go#L30 for nacos, https://github.com/go-kratos/kratos/blob/9662ef3c213d3f7802c27d6b85386bd649a76de7/contrib/registry/consul/registry.go#L21 for consul, etc...

Reply:
I will refactor it.